### PR TITLE
Various improvements for the p11test

### PR DESCRIPTION
**Pull request by Jakuje**
This fixes various issues in the p11test found mostly while testing over last months.

##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [ ] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

**Review by frankmorgner — APPROVED** [review]

File: src/tests/p11test/p11test_case_wrap.c [hunk 1]
```diff
@@ -280,6 +318,144 @@ static int test_wrap(test_cert_t *o, token_info_t *info, test_cert_t *key, test_
...
+		mech->params_len = sizeof(ctr_params);
+		break;
+	case CKM_AES_GCM:
+		mech->params = &gcm_params;
+		mech->params_len = sizeof(gcm_params);
+		break;
+	case CKM_AES_KEY_WRAP:
+	case:CKM_AES_KEY_WRAP_PAD: */
+		/* Nothing special ... */
+		break;
```
[the remark this answers is not in the document]
Comment (Jakuje) [reply src/tests/p11test/p11test_case_wrap.c]: Fixed

**Comment by Jakuje** [comment]
Rebased on top of master (squashing some of the style warnings) and added also a p11test as part of the CI run together with the PKCS11SPY to catch issues such as #2988 earlier.
But it fails now in Fedora so I guess it will need some more love somewhere.

**Review by frankmorgner — APPROVED** [review]

**Comment by Jakuje** [comment]
Now, we should see only failure in the valgrind as we are getting uninitialized memory from softhsm. Not sure what to do about that, but I think that can be handled later.

File: src/tests/p11test/p11test_case_wrap.c [hunk 2]
```diff
@@ -37,6 +38,103 @@ strip_pkcs7_padding(const unsigned char *message, unsigned long message_length,
...
+	}
+	fprintf(stderr, "\ncheck:\n");
+	for (int i = 0; i < check_len; i++) {
+		fprintf(stderr, ":%x", check[i]);
+	}
+	fprintf(stderr, "\n");
+	free(check);
+	return -1;
+}
+
```
Comment (xhanulik) [inline src/tests/p11test/p11test_case_wrap.c:137]: I would maybe add here note about the return values (as by `test_secret_sign_verify`).

Comment (Jakuje) [reply src/tests/p11test/p11test_case_wrap.c:146]: Thanks. Added a note and resolved the strict build issues reported.

**Comment by Jakuje** [comment]
I think this should be final now.

File: src/tests/p11test/p11test_case_wrap.c (hunk 2 above)
Comment (xhanulik) [reply src/tests/p11test/p11test_case_wrap.c:146]: Actually I meant the comment for `test_wrap`, as there `0` stands for ok/skip and `1` for error (sorry for misleading comment).

Comment (Jakuje) [reply src/tests/p11test/p11test_case_wrap.c:146]: Done too.

**Review by frankmorgner — APPROVED** [review]

**Review by xhanulik — APPROVED** [review]

**Comment by Jakuje** [comment]
Merging. Thanks for reviews!

**Pull request merged**